### PR TITLE
Get rid of rinoh_papersize

### PR DESCRIPTION
**Pull request by alexfargus**
I decided to tick off some of the items in #77. And started removing `rinoh_papersize`.

This appears to cause failures in the sphinx regression tests. I think this is because the default `paper_size` for `Book` is A4 but sphinx was ignoring this and using letter as the default - probably due to #178.

I see 2 options here:
1. Change the default `paper_size` variable in `Book` to `letter` and fix up any impacted tests.
2. Accept the breaking change and modify the regression tests to expect `A4` pages.

@brechtm what is your preferred solution?

### CODE REVIEW

File: src/rinoh/template.py [hunk 1]
```diff
@@ -497,8 +497,10 @@ def __init__(self, name, base=None, template=None, description=None,
         tmpl_cls = self.template
         for attr, value in options.items():
             options[attr] = tmpl_cls.validate_attribute(attr, value, True)
-        super().__init__(name, base=base or self.template, **options)
+        base = base or self.template
+        super().__init__(name, base=base, **options)
         self.description = description
+        self.variables.update(base.variables)
```
Comment (brechtm) [inline src/rinoh/template.py:503]: Why are you doing this? Variables are looked up in the base (see `RuleSet.get_variable()`) when not defined in the current template configuration. Note that your change overwrites this template configuration's variables with those of _base_.

**Comment by brechtm** [comment]
Look good to me except for my comment in `template.py`. Please also remove the corresponding documentation (`sphinx.rst`).

> I see 2 options here:
> 
>     1. Change the default `paper_size` variable in `Book` to `letter` and fix up any impacted tests.
>     2. Accept the breaking change and modify the regression tests to expect `A4` pages.

Since I'm European, I'd prefer option 2 :-)

**Comment by brechtm** [comment]
Perhaps also add a deprecation warning, like you did for _rinoh_stylesheet_ in your other PR?

File: src/rinoh/frontend/sphinx/__init__.py [hunk 2]
```diff
@@ -347,7 +342,6 @@ def setup(app):
     app.add_builder(RinohBuilder)
     app.add_config_value('rinoh_documents', default_documents, 'env')
     app.add_config_value('rinoh_stylesheet', None, 'html')
-    app.add_config_value('rinoh_paper_size', default_paper_size, 'html')
```
Comment (brechtm) [inline src/rinoh/frontend/sphinx/__init__.py:350]: The _default_paper_size_ function can be removed as well.

File: src/rinoh/template.py (hunk 1 above)
Comment (alexfargus) [reply src/rinoh/template.py]: This was my attempt to make the unit tests pass unmodified. 

From the information that you provided here it looks like the unit tests in `test_sphinx.py` should no longer be asserting that `paper_size` is in the `variables` dict. When I remove that assertion, the test passes and correctly falls back to `RuleSet.get_variable()` to determine `paper_size`.

File: src/rinoh/frontend/sphinx/__init__.py (hunk 2 above)
Comment (alexfargus) [reply src/rinoh/frontend/sphinx/__init__.py:340]: `default_paper_size` has been removed.

**Comment by alexfargus** [comment]
The expected pdfs for the sphinx regression tests have been updated to be A4. From a visual inspection of the original and new pdfs it appears that only the papersize has changed. This is the expected result of this breaking change.

**Comment by alexfargus** [comment]
@brechtm All open issues have been addressed and build is green. Please can you take another look.

**Comment by brechtm** [comment]
Thanks, @alexfargus. Merged! Because your PR consisted of two merged branches [1], I squashed your commits in cee3120910c2553675e007b1844783d116876b5c instead of merging them to keep things clear.

[1] See the image below: commits on the two branches have identical commit titles but different content making this very confusing. Please try to avoid this. 😄

image

**Pull request closed without merging**